tools/debian/convert_debian.py

```diff
@@ -34,97 +36,179 @@
     # Enumerate advisories + version info from security-tracker.
     with open(os.path.join(security_tracker_repo,
                            SECURITY_TRACKER_DSA_PATH)) as handle:
-        for line in handle:
-            line = line.rstrip()
-            if not line:
-                continue
-
-            if LEADING_WHITESPACE.match(line):
-                # Within current advisory.
-                if not current_advisory:
-                    raise ValueError('Unexpected tab.')
-
-                # {CVE-XXXX-XXXX CVE-XXXX-XXXX}
-                line = line.lstrip()
-                if line.startswith('{'):
-                    advisories[current_advisory].cves = line.strip('{}').split()
-                    continue
-
-                if line.startswith('NOTE:'):
-                    continue
-
-                version_match = VERSION_PATTERN.match(line)
-                if not version_match:
-                    raise ValueError('Invalid version line: ' + line)
-
-                advisories[current_advisory].versions.append(
-                    VersionInfo(version_match.group(1), version_match.group(2),
-                                version_match.group(3)))
-            else:
-                if line.strip().startswith('NOTE:'):
-                    continue
-
-                # New advisory.
-                dsa_match = DSA_PATTERN.match(line)
-                if not dsa_match:
-                    raise ValueError('Invalid line: ' + line)
-
-                current_advisory = dsa_match.group(1)
-                advisories[current_advisory] = AdvisoryInfo(
-                    current_advisory, dsa_match.group(2))
-
-    print(advisories)
+        parse_security_tracker_file(advisories, handle)
+
+    file_path_map = {}
+
+    for root, dirs, files in os.walk(os.path.join(webwml_repo, WEBWML_SECURITY_PATH)):
+        for file in files:
+            file_path_map[file] = os.path.join(root, file)
+
+    # Add descriptions to advisories from wml files
+    for key in advisories:
+        # TODO: use regex to remove suffix?
+        mapped_key_no_ext = key.lower().removesuffix('-1').removesuffix('-2').removesuffix('-3').removesuffix('-4')
+        val_wml = file_path_map.get(mapped_key_no_ext + ".wml")
+        val_data = file_path_map.get(mapped_key_no_ext + ".data")
+
+        if not val_wml:
+            print(mapped_key_no_ext)
+        else:
+            with open(val_wml) as handle:
+                data = handle.read()
+                html = WML_DESCRIPTION_PATTERN.findall(data)[0]
+                res = markdownify.markdownify(html)
+                advisories[key].details = res
+
+        if not val_data:
+            print(mapped_key_no_ext)
+        else:
+            with open(val_data) as handle:
+                data: str = handle.read()
+                report_date: str = WML_REPORT_DATE_PATTERN.findall(data)[0]
+                # Split by ',' here for the occasional case where there are two dates in the "publish" field
```

Comment:
Can you link to some examples? Curious which one is the right one.

Reply:
Turns out there was more than I originally thought.

Most of them is from 2006 - 2008, with the exception the one in 2013. The 2013 one has a extra dash after the date and no space between the comma and date, perhaps it's copied / pasted incorrectly.

```
2013-8-23,2013-9-01-
DSA-2740-2
2013-8-23,2013-9-01-
DSA-2740-1
2006-9-25, 2006-9-26
DSA-1184-2
2005-10-1, 2005-10-4
DSA-833-2
2008-2-11, 2008-2-12
DSA-1494-1
2007-10-2, 2007-10-12
DSA-1381-2
2007-3-20, 2007-03-28
DSA-1270-1
2006-12-3, 2006-12-3
DSA-1225-1
2006-11-30, 2006-12-1
DSA-1222-1
2006-11-20, 2006-12-27
DSA-1214
2006-11-12, 2006-11-13
DSA-1209
2006-11-9, 2006-11-19
DSA-1207-1
2006-11-2, 2006-12-1
DSA-1205-1
2006-9-28, 2006-10-2
DSA-1185-2
2006-9-3, 2006-10-13
DSA-1166
2006-8-29, 2006-9-13
DSA-1161
```

Reply:
I think the first date is the correct publihsed date. 

It looks like significant revisions result in a new date being added: https://salsa.debian.org/webmaster-team/webwml/-/commit/e1fa82c1bc3e984b17a65d980ffbb38b49f21087